tests/integration/loading/test_loading_component.py

```diff
@@ -0,0 +1,268 @@
+    def updateDiv(n_clicks):
+        if n_clicks is not None:
+            with lock:
+                return "changed 2"
+
+        return "content 2"
+
+    dash_dcc.start_server(app)
+
+    dash_dcc.wait_for_text_to_equal("#btn-1", "content 1")
+    dash_dcc.wait_for_text_to_equal("#btn-2", "content 2")
+
+    with lock:
+        dash_dcc.find_element("#btn-1").click()
+
+        dash_dcc.find_element(".loading-2 .dash-spinner")
+        dash_dcc.wait_for_text_to_equal("#btn-2", "")
+
+    dash_dcc.wait_for_text_to_equal("#btn-2", "changed 2")
+
+    with lock:
+        dash_dcc.find_element("#btn-2").click()
+
+        dash_dcc.find_element(".loading-1 .dash-spinner")
+        dash_dcc.wait_for_text_to_equal("#btn-1", "")
+
+    dash_dcc.wait_for_text_to_equal("#btn-1", "changed 1")
+
+    assert not dash_dcc.get_logs()
+
+
+def test_ldcp005_dynamic_loading_component(dash_dcc):
+    lock = Lock()
+
+    app = dash.Dash(__name__, suppress_callback_exceptions=True)
+
+    app.layout = html.Div([html.Button(id="btn-1"), html.Div(id="div-1")])
+
+    @app.callback(Output("div-1", "children"), [Input("btn-1", "n_clicks")])
+    def updateDiv(n_clicks):
+        if n_clicks is None:
+            return
+
+        with lock:
+            return html.Div(
+                [
+                    html.Button(id="btn-2"),
+                    dcc.Loading([html.Button(id="btn-3")], className="loading-1"),
+                ]
+            )
+
+    @app.callback(Output("btn-3", "children"), [Input("btn-2", "n_clicks")])
+    def updateDynamic(n_clicks):
+        if n_clicks is None:
+            return "content"
+
+        with lock:
+            return "changed"
+
+    dash_dcc.start_server(app)
+
+    dash_dcc.find_element("#btn-1")
+    dash_dcc.wait_for_text_to_equal("#div-1", "")
+
+    dash_dcc.find_element("#btn-1").click()
+
+    dash_dcc.find_element("#div-1 #btn-2")
+    dash_dcc.wait_for_text_to_equal("#btn-3", "content")
+
+    with lock:
+        dash_dcc.find_element("#btn-2").click()
+
+        dash_dcc.find_element(".loading-1 .dash-spinner")
+        dash_dcc.wait_for_text_to_equal("#btn-3", "")
+
+    dash_dcc.wait_for_text_to_equal("#btn-3", "changed")
+
+    assert not dash_dcc.get_logs()
+
+
+def test_ldcp006_children_identity(dash_dcc):
+    lock = Lock()
+
+    app = dash.Dash(__name__)
+    app.layout = html.Div([
+        html.Button("click", id="btn"),
+        dcc.Loading(dcc.Graph(id="graph"), className="loading")
+    ])
+
+    @app.callback(Output("graph", "figure"), [Input("btn", "n_clicks")])
+    def update_graph(n):
+        with lock:
+            bars = list(range(2, (n or 0) + 5))
+            return {
+                "data": [{"type": "bar", "x": bars, "y": bars}],
+                "layout": {"width": 400, "height": 400}
+            }
+
+    with lock:
+        dash_dcc.start_server(app)
+        dash_dcc.find_element(".loading .dash-spinner")
+        dash_dcc.find_element("#graph .js-plotly-plot")
```

Comment:
Could we also validate that the graph isn't visible?

Reply:
I did that with some of the [simpler items eg](https://github.com/plotly/dash-core-components/pull/740/files#diff-823b747b0b592ad60f40a9c017818677R219):
```
dash_dcc.wait_for_text_to_equal("#btn-3", "")
```
it looks like it's empty even though it actually still has text in it. Not sure exactly how to do that for a graph though - I suppose I could look directly at the `visibility` style on the element that sets it to `hidden`.

Reply:
If equivalent, all good :D